src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,661 @@
+            if image.ndim == 4:
+                num_images = image.shape[0]
+        elif isinstance(image, Image.Image):
+            W, H = image.size
+        else:
+            raise ValueError(f"Unsupported `image` type: {type(image)}.")
+
+        if num_images > 1 and output_prediction_format == "pil":
+            raise ValueError("`output_prediction_format='pil'` is not supported when passing multiple input images.")
+
+        # latents checks
+        if latents is not None:
+            if not torch.is_tensor(latents):
+                raise ValueError("`latents` must be a torch.FloatTensor.")
+            if not latents.dim() != 4:
+                raise ValueError(f"`latents` has unsupported dimensions or shape: {latents.shape}.")
+
+            if processing_resolution > 0:
+                max_orig = max(H, W)
+                new_H = H * processing_resolution // max_orig
+                new_W = W * processing_resolution // max_orig
+                if new_H == 0 or new_W == 0:
+                    raise ValueError(f"Extreme aspect ratio of the input image: [{W} x {H}]")
+                W, H = new_W, new_H
+            w = (W + self.vae_scale_factor - 1) // self.vae_scale_factor
+            h = (H + self.vae_scale_factor - 1) // self.vae_scale_factor
+            shape_expected = (num_images * ensemble_size, self.latent_space_size, h, w)
+
+            if latents.shape != shape_expected:
+                raise ValueError(f"`latents` has unexpected shape={latents.shape} expected={shape_expected}.")
+
+        # generator checks
+        if generator is not None:
+            device = self._execution_device
+            if isinstance(generator, torch.Generator):
+                if generator.device != device:
+                    raise ValueError("`generator` device differs from the pipeline's device.")
+            elif isinstance(generator, list):
+                if len(generator) != num_images * ensemble_size:
+                    raise ValueError(
+                        "The number generators must match the total number of ensemble members for all input images."
+                    )
+                if not all(g.device == device for g in generator):
+                    raise ValueError("At least one of the `generator` devices differs from the pipeline's device.")
+            else:
+                raise ValueError(f"Unsupported generator type: {type(generator)}.")
+
+    @torch.no_grad()
+    @replace_example_docstring(EXAMPLE_DOC_STRING)
+    def __call__(
+        self,
+        image: Union[Image.Image, np.ndarray, torch.FloatTensor],
+        num_inference_steps: Optional[int] = None,
+        ensemble_size: int = 1,
+        processing_resolution: Optional[int] = None,
+        match_input_resolution: bool = True,
+        resample_method_input: str = "bilinear",
+        resample_method_output: str = "bilinear",
+        batch_size: int = 1,
+        check_input: bool = True,
```

Comment:
what is this argument for? any reason we would allow not check input?

Reply:
It makes `MarigoldImageProcessor` check the input using the following method. It is not necessary most of the times, but it helps the user troubleshoot RGB data normalization issues when they pass data as `np.ndarray` or `torch.Tensor`. This is off by default to save a potentially costly check (when on CPU).
```
    @staticmethod
    def check_image_values_range(image: torch.FloatTensor) -> None:
        assert torch.is_floating_point(image)

        val_min = image.min().item()
        val_max = image.max().item()

        if val_min < -1.0 or val_max > 1.0:
            raise ValueError("Input image data is partially outside of the [-1,1] range.")
        if val_min >= 0.0:
            logger.warning(
                "Input image data is entirely in the [0,1] range; expecting [-1,1]. "
                "This could be an issue with normalization"
            )
```

Reply:
I think the responsibility of this kind of check should not be on the user 
can we set it as a config instead? The application can decide whether they want to run this validation step or not

Reply:
I moved this check into the preprocessor, it is performed when it is configured with `do_range_check`, which is True by default